### DISCUSSION TITLE
Start kodi in fullscreen (--fullscreen or alternative)

### DISCUSSION THREAD

**Discussion by rharmonson** [bakkeby/patches#58] [Q&A] [answered]
My objective is on start kodi 19 full screen.

When launching kodi --fullscreen it appears to open at the correct resolution 2160p, however, the first line is displayed at 1/2 the height of the statusbar, dwmlbocks. So half the statusbar is showing. Toggling the status bar hides the bar and results in a black band.

As an alternative to using "--fullscreen," I googled to see if there was a Rule isfullscreen like isfloating but not seeing it--I may have missed it.

I have an interim solution running kodi in a window, using TOGGLEFULLSCREEN and LOSEFULLSCREEN patches, then mod+y to toggle to full screen. It displays correctly using this method. 

Any guidance would be appreciated.

**Comment by bakkeby** [comment]
A stock dwm with no patches has support for applications starting in fullscreen based on the extended window manager hints (EWMH), this is handled in the `updatewindowtype` function, called from `manage`.


```c
void
updatewindowtype(Client *c)
{
	Atom state = getatomprop(c, netatom[NetWMState]);
	Atom wtype = getatomprop(c, netatom[NetWMWindowType]);

	if (state == netatom[NetWMFullscreen])
		setfullscreen(c, 1);
	if (wtype == netatom[NetWMWindowTypeDialog])
		c->isfloating = 1;
}
```

I tried to replicate this using `kodi --fullscreen` both in a stock dwm as well as my own build and I can't replicate this, even in a multi-monitor setup, so I suspect that there may be something else interfering.

> As an alternative to using "--fullscreen," I googled to see if there was a Rule isfullscreen like isfloating but not seeing it--I may have missed it.

I am not aware of any patch that does this, but it would not be particularly difficult to add. The result would be the same though as you would go through that same `setfullscreen` function both when the rule says that a client is in fullscreen and when the window has the `_NET_WM_STATE_FULLSCREEN` property set. It can be useful, however, for programs that don't offer the option of starting in fullscreen. I have something similar in my build.

**Reply by rharmonson** [reply]
If I understood you correctly, you were unable to replicate where kodi was shifted down 1/2 down the statusbar.

Sounds like I may have broke something. Strange things is that firefox-esr and other applications do full screen just fine. I should have shared that info.. sorry. :(

I have several assets so I will move to each of them in turn and verify it is reproduced on each for Kodi. If true, wipe one the assets and see if it reproduced with the minimum of installed components, patches, etc.

T`hank you, bakkeby.

**Reply by bakkeby** [reply]
I suppose the always center patch might interfere with this. Do you have that?

**Reply by rharmonson** [reply]
> I suppose the always center patch might interfere with this. Do you have that?

I do! Disabling and compiling..

**Comment by rharmonson** [comment, marked as answer]
You are the man!

Disabling always center patch fixed it.

Hm.. now the question is can I live without? Probably. lol

**Reply by bakkeby** [reply]
You can use the (original) ruled center patch, but I think it is just a matter of changing the order in which things happen (i.e. make the always center patch logic happen before it goes fullscreen, or have it ignore fullscreen windows).

I'll see if I can fix this in dwm-flexipatch.

**Reply by rharmonson** [reply]
I didn't mean to create work for you but that would be awesome.

Not familiar with github discussions. How do I mark this as answered/resolved?

**Reply by bakkeby** [reply]
I am no so familiar with discussions myself. As the author of the discussion / post you should have the option of selecting a comment (or part of a comment) as the accepted answer. This will be highlighted at the top for other users. Not all discussions have selected answers though.

**Reply by bakkeby** [reply]
I just pushed a change to dwm-flexipatch ref. [82a1276](https://github.com/bakkeby/dwm-flexipatch/commit/82a127630d4af24b587a575df50c564705c93e27) which is just moving that line setting fullscreen below the code of the alwayscenter/center patches. I don't foresee any unwanted side effects of this but let me know if you come across any.

**Reply by rharmonson** [reply]
Verified!

It is working beautifully. Thank you!
